### ISSUE TITLE
Discussion: Rename "Like" to "Heart"?

### ISSUE THREAD

**Issue by cinnamon-bun** [fraction/oasis#159]
**What's the problem you want solved?**
It feels inappropriate to "Like" someone's post about bad news but you may still want to show that you read the message and are supportive.

"Like" also feels Facebooky and bland.

**Bigger question**
Different SSB apps treat likes differently.  What do they mean?  Are they...

* ♻️ Retweets
* 🔝 Upvotes for the algorithm
* ❤️ Expression of support
* ⭐️ Bookmark, save, read later
* ✅ Read receipts, to show the author you saw their post

**Is there a solution you'd like to recommend?**
I wanted to rename Like to Heart because it feels warmer and less quantified.

But I'm realizing now: Someone in Oasis might like a ton of posts, not realizing they're spamming their feed with retweets when viewed in Patchwork.

How do the other SSB apps show likes?  On a separate page, or mixed into an author's own messages?

**Comment by christianbundy** [comment]
I'd be down with hearts. Might be useful to encode that in the message we're publishing, right now it's just `{type: "vote"}` but I think I've seen people encode "yup" / "like" / etc.

**Comment by cinnamon-bun** [comment]
A survey of prior art - this is the visual language our users already understand.  We don't want to copy this all, but whatever we do should use a familiar metaphor to avoid surprises

### Upvoting

Reddit
![reddit](https://user-images.githubusercontent.com/32660718/73794916-43e80b00-475e-11ea-81c9-e4782748e4a0.png)

Hacker News
![hn](https://user-images.githubusercontent.com/32660718/73795346-3c753180-475f-11ea-92ea-99e3b2d11f5a.png)

### Reblog, like, reply, share
Instagram
![instagram](https://user-images.githubusercontent.com/32660718/73794911-434f7480-475e-11ea-9685-3868d3f31927.png)

Twitter
![twitter](https://user-images.githubusercontent.com/32660718/73794917-43e80b00-475e-11ea-8d2f-21c739416d2c.png)

Tumblr
![tumblr](https://user-images.githubusercontent.com/32660718/73794918-43e80b00-475e-11ea-8ffc-e481c5ba03c4.png)

Facebook
![facebook](https://user-images.githubusercontent.com/32660718/73794913-43e80b00-475e-11ea-8ba4-cbc196e20111.png)

### Favorite for later
Weasyl
![weasyl](https://user-images.githubusercontent.com/32660718/73794914-43e80b00-475e-11ea-8768-030693935f87.png)

Furaffinity
![furaffinity](https://user-images.githubusercontent.com/32660718/73794915-43e80b00-475e-11ea-9a60-022fc7463ca7.png)

Artstation
![artstation](https://user-images.githubusercontent.com/32660718/73795105-b1943700-475e-11ea-92f4-301d4cbd3035.png)

Deviantart
![deviantart](https://user-images.githubusercontent.com/32660718/73795107-b1943700-475e-11ea-9b9c-6c687b0d0e76.png)

### Emoji reactions
Discord
![discord](https://user-images.githubusercontent.com/32660718/73795213-f61fd280-475e-11ea-8a15-db57fe6ce669.png)

Slack
![slack](https://user-images.githubusercontent.com/32660718/73795215-f6b86900-475e-11ea-8c51-b671a7b8d1a6.png)

Riot
![riot](https://user-images.githubusercontent.com/32660718/73795277-1780be80-475f-11ea-85a9-519f9fd63ea3.png)

**Comment by christianbundy** [comment]
Ramble: In the future I'd really like to add emoji-style reactions, but I think we'd need to have a solid spec and commitment from other clients to support it.

**Comment by cinnamon-bun** [comment]
I would **love** emoji reactions.  I like how playful and emotionally expressive they are.  Agreed that we'd need a spec first.

**Comment by georgeowell** [comment]
In favour of this!

**Comment by jedahan** [comment]
Love the idea. Heart is certainly an improvement. One thing I haven’t spent too much thinking about is a bit of tension between using familiar iconography which could lead to surprises when expectations and reality are different, versus completely alien definitions so no expectations are even created.

Reference is Mastodon “florp” vs Facebook Like. Could there be a word or icon that feels encouraging and familiar while being different enough that expectations are met? I love the mention of a vote being “an expression of support”. I’ll search through some Unicode and emoji for some possible examples.

**Comment by jedahan** [comment]
As far as emoji reactions, one idea I can capture in another issue - what if under a post we literally just listed all emoji found in people’s comments for a thread? Just scraped it from the message text as a sort of summary

**Comment by Powersource** [comment]
I think @staltz is very keen on adding emoji reactions to manyverse so I think that's gonna happen there at least. I think he mentioned he'd just change the `expression` field in the json to be the emoji, like `"expression": "❤️"` instead of`"expression": "Like"`, but correct me if I'm wrong.

**Comment by staltz** [comment]
@Powersource That's correct

**Comment by georgeowell** [comment]
I think we have settled on hearts no?

Emoji reactions is a different issue.

**Comment by christianbundy** [comment]
Yep, they're called "likes" in a few places throughout the code that we should change to "heart" and we should probably add the `expression` to the vote message.

**Comment by christianbundy** [comment]
Now that Manyverse has "reactions" maybe we should do that instead?

**Comment by johanbove** [comment]
I'm totally in for reactions! Similar to how Andre did it in ManyVerse. We'll need to see how it will render in the UI however. Popular posts with a lot of "reactions" will need to group similar reactions some how, like Github does it in their "issues", perhaps?

**Comment by Powersource** [comment]
I kind of like (although it could maybe be improved) how manyverse doesn't do any kind of numbering on the reactions and just puts them in a big pile. Feels a lot more like you're pouring love on someone.
